### ISSUE TITLE
Jabber plugin can't connect

### ISSUE THREAD

**Issue by xf0e** [licq-im/licq#30]
After upgrade to version 1.7.1 i can't connect to my jabber account anymore. I always get: 
[ERR] jabber: connection closed by the server.

My account is fine, since i can connect with other clients.

**Comment by erijo** [comment]
Did it work with 1.7.0?

**Comment by xf0e** [comment]
I think so, but I'm not sure.

**Comment by erijo** [comment]
Can you provide some more details about your system, how you installed licq etc.

Can you post the log when running licq -d 31 and trying to connect?

**Comment by xf0e** [comment]
Thank you for you help. Here is a requested log from licq with a clean profile. I'm on a archlinux and licq package is based on official repo package with added jabber support.
http://pastebin.com/se2PJ8rt

**Comment by erijo** [comment]
I created an account on jabber.ru and I could connect with 1.7.1.

Could you pleas try running licq with a new basedir: licq -d 31 -b tmp-basedir
Then add your account and see if you can connect. Please post the log output in either case.

**Comment by xf0e** [comment]
Hello. With tmp-basedir i geet the same behavior like on the clean profile above. Here is the log:
http://pastebin.com/cQcmAD53

**Comment by lampaczcz** [comment]
Hello,

i've same problem. Using licq version 1.7.1, compiled on Jan 31 2013 on Debian GNU/Linux jessie/sid (testing)

Here is log from licq -d 31 -b tmp-basedir 
http://pastebin.com/apVz095h

**Comment by erijo** [comment]
I think this is a bug in gloox, see http://bugs.camaya.net/ticket/?id=202

Can you test with gloox version 1.0.2?

**Comment by xf0e** [comment]
Hello, i just recompiled latest licq with gloox 1.0.2 and it's not working. Same error: 23:44:07.220 [ERR] jabber: connection closed by the server

**Comment by erijo** [comment]
Thanks for testing with latest gloox!

Just to be sure, can you please verify that the correct gloox version was used at runtime by checking the Licq log for the line "Using gloox version 1.x.y"?

**Comment by xf0e** [comment]
Yes, i can :D Here is the connection log: http://pastebin.com/bnd17V8h

However the client is saying that version 1.0.1 is used and same information is in glooxversion.h

tail -n1  /usr/include/gloox/glooxversion.h
#define GLOOXVERSION 0x010001,

but: pacman -Q | grep gloox
gloox 1.0.2-1

anyway this is at least 1.0.1

**Comment by xf0e** [comment]
Just downloaded latest version from http://camaya.net/download/gloox-1.0.2.tar.bz2 and in glooxversion.h is version 1.0.1 defined.

**Comment by erijo** [comment]
Could you please try the example program that I posted in this bug report: https://bugzilla.redhat.com/show_bug.cgi?id=909859

Just to see if the problem in gloox is still there.

**Comment by xf0e** [comment]
no error:
g++ -o tagtest ttttt.cpp -lgloox && ./tagtest
echo $?
0

**Comment by xf0e** [comment]
I also compiled the 1.7.0 version and the same error is already there. Version 1.6.1 can't be compiled anymore on my current setup because of error in pipe.cpp.

**Comment by erijo** [comment]
Just to be sure, can you try compiling the tagtest program again, but now with -O2 to see that the bug doesn't appear when enabling optimization.

I created an account on jabbim.com and I'm able to connect with gloox 1.0.2 + licq 1.7. There must be something in our set up that differs. Are you running on 32-bit or 64-bit (see output from uname -a)? Which compiler flags do you use when building gloox and licq?

Can you try the message_example available in the gloox source? Edit src/examples/message_example.cpp and change hurkhurk@example.net to yourid@jabbim.com and hurkhurks to your password. Build gloox and then run ./src/examples/message_example.

**Comment by xf0e** [comment]
I'm on archlinux 64bit, GCC 4.8.1 and compiler flags are: -march=native -mtune=native -O2 -pipe -fstack-protector --param=ssp-buffer-size=4 -D_FORTIFY_SOURCE=2.  I was also playing with compiles flags, and get the same results 

message_example test: http://pastebin.com/JxyWx433

I created 2 more accounts on different servers and can't connect to any of them with licq. However psi can connect to all of them.

**Comment by erijo** [comment]
The output from message_example hangs stops at the same place as it does for Licq, so I think we can be fairly certain that the problem lies within gloox. For reference, this is how it looks for me when running message_example: http://pastebin.com/sV2sJLu1

I think we should report this to bug tracker for gloox (http://bugs.camaya.net/). It would probably be best if you could do it, but I can do it if you want.

**Comment by xf0e** [comment]
I would appreciate if you could send a bug report since you're sure know more about gloox. I'm not sure how I should describe the problem on the bug tracker.

**Comment by erijo** [comment]
I have submitted the bug: http://bugs.camaya.net/ticket/?id=207

It would be good if you could add yourself as watcher.

**Comment by cicku** [comment]
Hi,

I'm a packager from Fedora Project, we currently ships 1.0 version of gloox, should I rebuild licq against gloox 1.0.2?

Thanks.

**Comment by erijo** [comment]
gloox 1.0.2 should have fixed at least one connection problem, so it would be good to rebuild against 1.0.2.

But there seems to be another one lurking that I reported upstream. That one hasn't been fixed yet.

**Comment by cicku** [comment]
Former maintainer said you can switch to jreen.

http://qutim.org/jreen

Thanks.

**Comment by kolp** [comment]
Could someone please test licq 1.8.0 + gloox 1.0.6? It appears to work.

**Comment by xf0e** [comment]
Just compiled licq 1.8.0 against gloox 1.0.6. Still get same results - i'm not able to connect to any of my jabber accounts.  Here is the log: http://pastebin.com/3GxzGt8G sometimes it's look like this: http://pastebin.com/8Ju4v07k
